src/main/java/org/interledger/codecs/oer/OerUint64Codec.java

```diff
@@ -33,12 +35,14 @@ public OerUint64 read(final CodecContext context, final InputStream inputStream)
     Objects.requireNonNull(context);
     Objects.requireNonNull(inputStream);
 
-    long value = 0;
-    for (int i = 0; i < 8; i++) {
-      value <<= Byte.SIZE;
-      value |= (inputStream.read() & 0xFF);
+    byte[] value = new byte[8];
+    int read = inputStream.read(value);
+    
+    if (read != 8) {
+      throw new IOException("unexpected end of stream. expected 8 bytes, read " + read);
```

Comment:
Oops, my bad - I was wrong. But this might be preferable to help the reader not exceed 8 bytes:

```
int read = inputStream.read(value, 0, 8);
if (read != 8) {
    throw new IOException("unexpected end of stream. expected 8 bytes, read " + read);
}

```